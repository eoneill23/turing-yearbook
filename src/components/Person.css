.personCard {
  background-color: white;
  border-radius: 5px;
  margin: 2%;
  padding: 20px;
  width: 15%;
}

img {
  border-radius: 5px;
  height: 250px;
  width: 200px;
}