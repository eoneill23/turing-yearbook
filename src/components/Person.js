import React from 'react';
import './Person.css';
import { tsPropertySignature } from '@babel/types';

let Person = ({name, quote, superlative, photo, id, deleteStudent, type}) => {
  let button;

  if (type === 'students') {
    button = <button onClick={() => deleteStudent(id, type)} type={type}>Delete</button>
  } else {
    button = ''
  }

  return (
    <section className="personCard" key={id}>
      <img src={photo}></img>
      <p className="personName">{name}</p>
      <p className="personQuote">"{quote}"</p>
      <p className="personSaying">{superlative}</p>
      {button}
    </section>
  )
}

export default Person;