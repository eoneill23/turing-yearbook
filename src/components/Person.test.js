import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person.js';

describe('Person', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Person 
      name={'Travis'}
      quote={'Cheers'}
      superlative={'Most likely to say cheers'}
      photo={''}
      id={1}
      deleteStudent={jest.fn()}
      type={'students'}
    />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if students is passed as the type prop', () => {
    const wrapper = shallow(<Person 
      name={'Travis'}
      quote={'Cheers'}
      superlative={'Most likely to say cheers'}
      photo={''}
      id={1}
      deleteStudent={jest.fn()}
      type={'students'}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call the delete student prop when the delete button is clicked', () => {
    const deleteStudentMock = jest.fn();
    const wrapper = shallow(<Person 
      name={'Travis'}
      quote={'Cheers'}
      superlative={'Most likely to say cheers'}
      photo={''}
      id={1}
      deleteStudent={deleteStudentMock}
      type={'students'}
    />)

    wrapper.find('button').simulate('click');

    expect(deleteStudentMock).toHaveBeenCalled();
  })
});