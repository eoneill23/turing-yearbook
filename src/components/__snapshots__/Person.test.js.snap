// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot if students is passed as the type prop 1`] = `
<section
  className="personCard"
  key="1"
>
  <img
    src=""
  />
  <p
    className="personName"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    Travis
  </p>
  <p
    className="personQuote"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    "
    Cheers
    "
  </p>
  <p
    className="personSaying"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    Most likely to say cheers
  </p>
  <button
    onClick={[Function]}
    type="students"
  >
    Delete
  </button>
</section>
`;

exports[`Person should match the snapshot with all data passed in correctly 1`] = `
<section
  className="personCard"
  key="1"
>
  <img
    src=""
  />
  <p
    className="personName"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    Travis
  </p>
  <p
    className="personQuote"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    "
    Cheers
    "
  </p>
  <p
    className="personSaying"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    Most likely to say cheers
  </p>
  <button
    onClick={[Function]}
    type="students"
  >
    Delete
  </button>
</section>
`;
